src/parser/de.rs

```diff
@@ -0,0 +1,309 @@
+use std::collections::HashMap;
+
+use serde::de::{DeserializeOwned, Error};
+use serde_json::{Map, Value};
+
+use crate::async_graphql_hyper::GraphQLRequestLike;
+
+type PosValHolder = HashMap<String, Vec<(String, String)>>;
+
+#[derive(Debug, serde::Deserialize, Default)]
+pub struct Parser {
+  root: Option<String>,
```

Comment:
```suggestion
  operation: String,
```
Operation can't be optional